File: HMS_py/modLogin.cs
//==========================================
// Login Module for HMS (C# Port)
// Reads Analysis.ini and handles authentication
//==========================================

using System;
using System.IO;
using System.Data.SqlClient;

namespace HMS
{
    //==========================================
    // Login Configuration
    //==========================================
    public struct LoginConfig
    {
        public string Server;
        public string Database;
        public string CompanyCode;
        public string CompanyName;
        public string ReportsPath;
        public string TempPath;
        public string ImagePath;
        public string BackupPath;
        public string UserID;
        public string Password;
    }

    public class modLogin
    {
        public static LoginConfig gLoginConfig = new LoginConfig();
        public static string gLoggedInUser = "";
        public static string gLoggedInCompany = "";
        public static string gConnectionString = "";

        //==========================================
        // Read Analysis.ini file
        //==========================================
        public static bool ReadAnalysisIni()
        {
            string sFilePath = AppDomain.CurrentDomain.BaseDirectory + "HMS2425\\Analysis.ini";

            try
            {
                if (!File.Exists(sFilePath))
                {
                    System.Windows.Forms.MessageBox.Show("Analysis.ini not found!", 
                        "Error", System.Windows.Forms.MessageBoxButtons.OK, 
                        System.Windows.Forms.MessageBoxIcon.Exclamation);
                    return false;
                }

                string[] lines = File.ReadAllLines(sFilePath);
                
                foreach (string sLine in lines)
                {
                    string line = sLine.Trim();
                    if (line.StartsWith("[")) continue;

                    if (line.IndexOf("=") > 0)
                    {
                        string[] arrParts = line.Split('=');
                        string key = arrParts[0].Trim();
                        string value = arrParts[1].Trim();

                        switch (key)
                        {
                            case "1":
                                gLoginConfig.Server = value;
                                break;
                            case "2":
                                gLoginConfig.ReportsPath = value;
                                break;
                            case "3":
                                gLoginConfig.TempPath = value;
                                break;
                            case "5":
                                gLoginConfig.BackupPath = value;
                                break;
                            case "6":
                                gLoginConfig.Database = value;
                                break;
                            case "7":
                                gLoginConfig.CompanyCode = value;
                                break;
                            case "8":
                                gLoginConfig.CompanyName = value;
                                break;
                            case "10":
                                gLoginConfig.ImagePath = value;
                                break;
                        }
                    }
                }

                return true;
            }
            catch (Exception ex)
            {
                System.Windows.Forms.MessageBox.Show("Error reading Analysis.ini: " + ex.Message,
                    "Error", System.Windows.Forms.MessageBoxButtons.OK,
                    System.Windows.Forms.MessageBoxIcon.Exclamation);
                return false;
            }
        }

        //==========================================
        // Get Company List from Database
        //==========================================
        public static string GetCompanyList()
        {
            string sCon = GetConnectionString();
            string sList = "";

            try
            {
                using (SqlConnection conn = new SqlConnection(sCon))
                {
                    conn.Open();

                    string query = "SELECT SubCode, Name FROM SubGroup WHERE GroupCode='C' ORDER BY Name";
                    using (SqlCommand cmd = new SqlCommand(query, conn))
                    {
                        using (SqlDataReader dr = cmd.ExecuteReader())
                        {
                            while (dr.Read())
                            {
                                sList += dr["SubCode"].ToString() + "|" + dr["Name"].ToString() + ",";
                            }
                        }
                    }
                }

                if (sList.EndsWith(","))
                {
                    sList = sList.Substring(0, sList.Length - 1);
                }

                return sList;
            }
            catch (Exception)
            {
                return "";
            }
        }

        //==========================================
        // Get Connection String
        //==========================================
        public static string GetConnectionString()
        {
            //Use SQL Server authentication
            string sCon = "Provider=SQLOLEDB;Data Source=" + gLoginConfig.Server + 
                         ";Initial Catalog=" + gLoginConfig.Database + 
                         ";User ID=sa;Password=sql2005;";

            gConnectionString = sCon;
            return sCon;
        }

        //==========================================
        // Get Company Code from Name
        //==========================================
        public static string GetCompanyCode(string sCompanyName)
        {
            string sCode = sCompanyName;

            if (sCompanyName.IndexOf("|") > 0)
            {
                string[] arrTemp = sCompanyName.Split('|');
                if (arrTemp.Length >= 1)
                {
                    sCode = arrTemp[0];
                }
            }

            return sCode;
        }

        //==========================================
        // Authenticate User
        //==========================================
        public static bool AuthenticateUser(string sUserCode, string sPassword)
        {
            string sCon = GetConnectionString();

            try
            {
                using (SqlConnection conn = new SqlConnection(sCon))
                {
                    conn.Open();

                    string query = "SELECT UserName, Password FROM UserMast WHERE UserCode=@UserCode";
                    using (SqlCommand cmd = new SqlCommand(query, conn))
                    {
                        cmd.Parameters.AddWithValue("@UserCode", sUserCode);

                        using (SqlDataReader dr = cmd.ExecuteReader())
                        {
                            if (dr.Read())
                            {
                                string dbPassword = dr["Password"].ToString().Trim();

                                //Compare password (plain text or encrypted based on system)
                                if (dbPassword == sPassword)
                                {
                                    gLoggedInUser = dr["UserName"].ToString();
                                    return true;
                                }
                            }
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                System.Windows.Forms.MessageBox.Show("Error authenticating user: " + ex.Message,
                    "Error", System.Windows.Forms.MessageBoxButtons.OK,
                    System.Windows.Forms.MessageBoxIcon.Exclamation);
            }

            return false;
        }

        //==========================================
        // Validate Login
        //==========================================
        public static bool ValidateLogin(string sUserCode, string sPassword, ref string sMessage)
        {
            sMessage = "";

            //Check empty fields
            if (sUserCode.Trim() == "")
            {
                sMessage = "Please enter User Code";
                return false;
            }

            if (sPassword.Trim() == "")
            {
                sMessage = "Please enter Password";
                return false;
            }

            //Authenticate
            if (!AuthenticateUser(sUserCode, sPassword))
            {
                sMessage = "Invalid User or Password";
                return false;
            }

            return true;
        }

        //==========================================
        // Get User Rights
        //==========================================
        public static System.Data.DataTable GetUserRights(string sUserCode)
        {
            System.Data.DataTable dt = new System.Data.DataTable();
            string sCon = GetConnectionString();

            try
            {
                using (SqlConnection conn = new SqlConnection(sCon))
                {
                    conn.Open();

                    string query = "SELECT ModuleName, Rights FROM UserRights WHERE UserCode=@UserCode";
                    using (SqlCommand cmd = new SqlCommand(query, conn))
                    {
                        cmd.Parameters.AddWithValue("@UserCode", sUserCode);

                        using (SqlDataAdapter da = new SqlDataAdapter(cmd))
                        {
                            da.Fill(dt);
                        }
                    }
                }
            }
            catch (Exception ex)
            {
                System.Windows.Forms.MessageBox.Show("Error loading user rights: " + ex.Message,
                    "Error", System.Windows.Forms.MessageBoxButtons.OK,
                    System.Windows.Forms.MessageBoxIcon.Exclamation);
            }

            return dt;
        }
    }
}

File: HMS_py/modLogin.vb
'==========================================
' Login Module for HMS (VB.NET Port)
' Reads Analysis.ini and handles authentication
'==========================================

Imports System.IO
Imports System.Data.SqlClient

Public Module modLogin

    '==========================================
    ' Login Configuration Type
    '==========================================
    Public Structure LoginConfig
        Public Server As String
        Public Database As String
        Public CompanyCode As String
        Public CompanyName As String
        Public ReportsPath As String
        Public TempPath As String
        Public ImagePath As String
        Public BackupPath As String
        Public UserID As String
        Public Password As String
    End Structure

    Public gLoginConfig As LoginConfig
    Public gLoggedInUser As String
    Public gLoggedInCompany As String
    Public gConnectionString As String

    '==========================================
    ' Read Analysis.ini file
    '==========================================
    Public Function ReadAnalysisIni() As Boolean
        Dim sFilePath As String
        Dim sLine As String

        Try
            sFilePath = AppDomain.CurrentDomain.BaseDirectory & "HMS2425\Analysis.ini"

            If Not File.Exists(sFilePath) Then
                MsgBox("Analysis.ini not found!", vbExclamation)
                Return False
            End If

            Using sr As New StreamReader(sFilePath)
                Do While Not sr.EndOfStream
                    sLine = Trim(sr.ReadLine())
                    If Left(sLine, 1) = "[" Then GoTo NextLine

                    If InStr(sLine, "=") > 0 Then
                        Dim arrParts() As String = Split(sLine, "=")
                        Dim key As String = Trim(arrParts(0))
                        Dim value As String = Trim(arrParts(1))

                        Select Case key
                            Case "1"
                                gLoginConfig.Server = value
                            Case "2"
                                gLoginConfig.ReportsPath = value
                            Case "3"
                                gLoginConfig.TempPath = value
                            Case "5"
                                gLoginConfig.BackupPath = value
                            Case "6"
                                gLoginConfig.Database = value
                            Case "7"
                                gLoginConfig.CompanyCode = value
                            Case "8"
                                gLoginConfig.CompanyName = value
                            Case "10"
                                gLoginConfig.ImagePath = value
                        End Select
                    End If
NextLine:
                Loop
            End Using

            Return True

        Catch ex As Exception
            MsgBox("Error reading Analysis.ini: " & ex.Message, vbExclamation)
            Return False
        End Try
    End Function

    '==========================================
    ' Get Company List from Database
    '==========================================
    Public Function GetCompanyList() As String
        Dim sCon As String = GetConnectionString()
        Dim sList As String = ""

        Try
            Using conn As New SqlConnection(sCon)
                conn.Open()

                Using cmd As New SqlCommand("SELECT SubCode, Name FROM SubGroup WHERE GroupCode='C' ORDER BY Name", conn)
                    Using dr As SqlDataReader = cmd.ExecuteReader()
                        Do While dr.Read()
                            sList = sList & dr("SubCode").ToString() & "|" & dr("Name").ToString() & ","
                        Loop
                    End Using
                End Using
            End Using

            If Right(sList, 1) = "," Then
                sList = Left(sList, Len(sList) - 1)
            End If

            Return sList

        Catch ex As Exception
            Return ""
        End Try
    End Function

    '==========================================
    ' Get Connection String
    '==========================================
    Public Function GetConnectionString() As String
        Dim sCon As String

        'Use SQL Server authentication
        sCon = "Provider=SQLOLEDB;Data Source=" & gLoginConfig.Server & _
              ";Initial Catalog=" & gLoginConfig.Database & _
              ";User ID=sa;Password=sql2005;"

        gConnectionString = sCon
        Return sCon
    End Function

    '==========================================
    ' Get Company Code from Name
    '==========================================
    Public Function GetCompanyCode(sCompanyName As String) As String
        Dim sCode As String = sCompanyName

        If InStr(sCompanyName, "|") > 0 Then
            Dim arrTemp() As String = Split(sCompanyName, "|")
            If UBound(arrTemp) >= 0 Then
                sCode = arrTemp(0)
            End If
        End If

        Return sCode
    End Function

    '==========================================
    ' Authenticate User
    '==========================================
    Public Function AuthenticateUser(sUserCode As String, sPassword As String) As Boolean
        Dim sCon As String = GetConnectionString()

        Try
            Using conn As New SqlConnection(sCon)
                conn.Open()

                Using cmd As New SqlCommand("SELECT UserName, Password FROM UserMast WHERE UserCode=@UserCode", conn)
                    cmd.Parameters.AddWithValue("@UserCode", sUserCode)

                    Using dr As SqlDataReader = cmd.ExecuteReader()
                        If dr.Read() Then
                            Dim dbPassword As String = dr("Password").ToString().Trim()

                            'Compare password (plain text or encrypted based on system)
                            If dbPassword = sPassword Then
                                gLoggedInUser = dr("UserName").ToString()
                                Return True
                            End If
                        End If
                    End Using
                End Using
            End Using

        Catch ex As Exception
            MsgBox("Error authenticating user: " & ex.Message, vbExclamation)
        End Try

        Return False
    End Function

    '==========================================
    ' Validate Login
    '==========================================
    Public Function ValidateLogin(sUserCode As String, sPassword As String, ByRef sMessage As String) As Boolean
        sMessage = ""

        'Check empty fields
        If Trim(sUserCode) = "" Then
            sMessage = "Please enter User Code"
            Return False
        End If

        If Trim(sPassword) = "" Then
            sMessage = "Please enter Password"
            Return False
        End If

        'Authenticate
        If Not AuthenticateUser(sUserCode, sPassword) Then
            sMessage = "Invalid User or Password"
            Return False
        End If

        Return True
    End Function

    '==========================================
    ' Get User Rights
    '==========================================
    Public Function GetUserRights(sUserCode As String) As DataTable
        Dim dt As New DataTable()
        Dim sCon As String = GetConnectionString()

        Try
            Using conn As New SqlConnection(sCon)
                conn.Open()

                Using cmd As New SqlCommand("SELECT ModuleName, Rights FROM UserRights WHERE UserCode=@UserCode", conn)
                    cmd.Parameters.AddWithValue("@UserCode", sUserCode)

                    Using da As New SqlDataAdapter(cmd)
                        da.Fill(dt)
                    End Using
                End Using
            End Using

        Catch ex As Exception
            MsgBox("Error loading user rights: " & ex.Message, vbExclamation)
        End Try

        Return dt
    End Function

End Module